.github/workflows/test.yml

```diff
@@ -0,0 +1,186 @@
+          chmod u+x FlexFringe/flexfringe
+          mv spdfa-config.ini FlexFringe/ini/
+
+      - name: Copy the test file in the top directory
+        shell: bash
+        run: |
+          cd SAGE-updated/
+          mv test-scripts/* ..
+
+      ################################################################
+      #############    RUN BOTH VERSIONS OF SAGE     #################
+      ################################################################
+
+      - name: Run SAGE on the main branch
+        if: '!contains(github.event.pull_request.labels.*.name, ''changes-ags'')'   # With 'changes-ags' label no regression tests will be run
+        shell: bash
+        run: |
+          cd SAGE-main/
+          echo "Running CPTC-2017..."
+          python sage.py alerts/cptc-2017/ orig-2017 --dataset cptc --keep-files
+          echo "Running CPTC-2018..."
+          python sage.py alerts/cptc-2018/ orig-2018 --dataset cptc --keep-files
+          echo "Running CCDC-2018..."
+          python sage.py alerts/ccdc/ orig-ccdc --dataset other --keep-files
+
+          cp -R orig-2017.txt orig-2017.txt.ff.final.json orig-2017.txt.ff.finalsinks.json orig-2017AGs/ ../  # Might have to be updated if FlexFringe changes the names
+          cp -R orig-2018.txt orig-2018.txt.ff.final.json orig-2018.txt.ff.finalsinks.json orig-2018AGs/ ../
+          cp -R orig-ccdc.txt orig-ccdc.txt.ff.final.json orig-ccdc.txt.ff.finalsinks.json orig-ccdcAGs/ ../
+
+      - name: Run SAGE on the updated branch
+        shell: bash
+        run: |
+          cd SAGE-updated/
+          echo "Running CPTC-2017..."
+          python sage.py alerts/cptc-2017 updated-2017 --dataset cptc --keep-files
+          echo "Running CPTC-2018..."
+          python sage.py alerts/cptc-2018/ updated-2018 --dataset cptc --keep-files
+          echo "Running CCDC-2018..."
+          python sage.py alerts/ccdc/ updated-ccdc --dataset other --keep-files
+
+          cp -R updated-2017.txt updated-2017.txt.ff.final.json updated-2017.txt.ff.finalsinks.json updated-2017AGs/ ../  # Might have to be updated if FlexFringe changes the names
+          cp -R updated-2018.txt updated-2018.txt.ff.final.json updated-2018.txt.ff.finalsinks.json updated-2018AGs/ ../
+          cp -R updated-ccdc.txt updated-ccdc.txt.ff.final.json updated-ccdc.txt.ff.finalsinks.json updated-ccdcAGs/ ../
+
+
+      ################################################################
+      #################    RUN REGRESSION TESTS    ###################
+      ################################################################
+
+      - name: Run regression tests on CPTC-2017
+        env:
+          CHANGES_IDS: ${{ contains(github.event.pull_request.labels.*.name, 'changes-ids') }}  # With 'changes-ids' label state IDs will be removed from the AGs
+        if: '!contains(github.event.pull_request.labels.*.name, ''changes-ags'')'  # With 'changes-ags' label no regression tests will be run
```

Comment:
So there are three cases:
1. Label `changes-ags` is present ⇒ don't run the regression tests
2. Label `changes-ids` is present ⇒ run with `-i` (removes IDs and skips `stats-sinks-ags.sh`, since the states cannot be matched due to different IDs)
3. Else, run normally

Something like this (tested on my fork):
![image](https://github.com/tudelft-cda-lab/SAGE/assets/45541174/e08af3df-bb56-4c3b-88ad-85361a2c387a)